### PR TITLE
Issues/33 set user institution in keycloack deployment

### PR DESCRIPTION
**Pull request by kostobog**
PR for issue #33 
Only the last commit [64cfe2c](https://github.com/kbss-cvut/record-manager/pull/35/commits/64cfe2c2bde7d680cbbfcbdf0afff5803a0ccce6) is relevant, see file changes in file `OidcUserController`. The other commits are from PR #31.

### CODE REVIEW

File: src/main/java/cz/cvut/kbss/study/rest/OidcUserController.java [hunk 1]
```diff
@@ -61,6 +63,36 @@ public List<User> getUsers(@RequestParam(value = "institution", required = false
...
+        List<URI> institutionURIs = Arrays.asList(user.getInstitution(), original.getInstitution())
+                .stream().map(i -> i != null ? i.getUri() : null).collect(Collectors.toList());
+        if (Objects.equals(institutionURIs.get(0), institutionURIs.get(1))){
+            LOG.warn("Ignoring attempt to add user {} to institution {} because it is already in institution {}.", user, institutionURIs.get(0), institutionURIs.get(1));
+            return;
+        }
+
+        // make sure only institution is updated
+        Institution newInstitution = user.getInstitution();
+        user = original.copy();
```
Comment (blcham) [inline src/main/java/cz/cvut/kbss/study/rest/OidcUserController.java:88]: I would use original directly

File: src/main/java/cz/cvut/kbss/study/rest/OidcUserController.java [hunk 2]
```diff
@@ -61,6 +63,36 @@ public List<User> getUsers(@RequestParam(value = "institution", required = false
...
+                                      @RequestParam(value = "email", defaultValue = "true") boolean sendEmail) {
+        if (!username.equals(user.getUsername())) {
+            throw new BadRequestException("The passed user's username is different from the specified one.");
+        }
+        final User original = getByUsername(username);
+
+        assert original != null;
+
+        // validate institution update is valid
+        List<URI> institutionURIs = Arrays.asList(user.getInstitution(), original.getInstitution())
```
Comment (blcham) [inline src/main/java/cz/cvut/kbss/study/rest/OidcUserController.java:79]: i guess we do not need this

**Review by blcham — CHANGES_REQUESTED** [review]

File: src/main/java/cz/cvut/kbss/study/rest/OidcUserController.java (hunk 1 above)
Comment (kostobog) [reply src/main/java/cz/cvut/kbss/study/rest/OidcUserController.java]: Fixed in commit [3402342](https://github.com/kbss-cvut/record-manager/pull/35/commits/3402342481154547dc4c4f81c228f92d513013ec)

File: src/main/java/cz/cvut/kbss/study/rest/OidcUserController.java (hunk 2 above)
Comment (kostobog) [reply src/main/java/cz/cvut/kbss/study/rest/OidcUserController.java]: @blcham removed in commit [3402342](https://github.com/kbss-cvut/record-manager/pull/35/commits/3402342481154547dc4c4f81c228f92d513013ec)

File: src/main/resources/model.ttl [hunk 3]
```diff
@@ -163,5 +169,29 @@ rm:user rdf:type owl:Class ;
...
+
+###  http://onto.fel.cvut.cz/ontologies/record-manager/open-record-phase
+rm:open-record-phase rdf:type owl:Class ;
+        rdfs:subClassOf rm:record-phase ;
+        rdfs:label "open record state"@en .
+
+###  http://onto.fel.cvut.cz/ontologies/record-manager/valid-record-phase
+rm:valid-record-phase rdf:type owl:Class ;
+        rdfs:subClassOf rm:record-phase ;
+        rdfs:label "valid record state"@en .
```
Comment (blcham) [inline src/main/resources/model.ttl:185]: I am confused now what will happen if i try to merge this ... I think it was already resolved somewhere.

**Review by blcham — CHANGES_REQUESTED** [review]
I can still see "state", not sure what to do with the request.

File: src/main/resources/model.ttl (hunk 3 above)
Comment (kostobog) [reply src/main/resources/model.ttl:185]: I can make a new pull request.

**Review by blcham — APPROVED** [review]

**Pull request merged**